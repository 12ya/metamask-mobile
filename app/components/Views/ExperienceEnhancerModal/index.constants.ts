export default {
  LINK_BUTTON: 'experience-enhancer-modal-link-button',
  CANCEL_BUTTON: 'experience-enhancer-modal-cancel-button',
  ACCEPT_BUTTON: 'experience-enhancer-modal-accept-button',
};
