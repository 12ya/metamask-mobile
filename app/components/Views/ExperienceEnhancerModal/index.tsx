import React, { useRef } from 'react';
import { View } from 'react-native';

import { strings } from '../../../../locales/i18n';
import BottomSheet, {
  BottomSheetRef,
} from '../../../component-library/components/BottomSheets/BottomSheet';
import Text, {
  TextVariant,
} from '../../../component-library/components/Texts/Text';
import createStyles from './styles';
import BottomSheetFooter, {
  ButtonsAlignment,
} from '../../../component-library/components/BottomSheets/BottomSheetFooter';
import Button, {
  ButtonSize,
  ButtonVariants,
} from '../../../component-library/components/Buttons/Button';
import { ButtonProps } from '../../../component-library/components/Buttons/Button/Button.types';
import { setDataCollectionForMarketing } from '../../../actions/security';

const ExperienceEnhancerModal = () => {
  const styles = createStyles();
  const bottomSheetRef = useRef<BottomSheetRef>(null);

  const cancelButtonProps: ButtonProps = {
    variant: ButtonVariants.Secondary,
    label: strings('experience_enhancer_modal.cancel'),
    size: ButtonSize.Lg,
    onPress: () => {
      setDataCollectionForMarketing(false);
      bottomSheetRef.current?.onCloseBottomSheet();
    },
  };

  const acceptButtonProps: ButtonProps = {
    variant: ButtonVariants.Primary,
    label: strings('experience_enhancer_modal.accept'),
    size: ButtonSize.Lg,
    onPress: () => {
      setDataCollectionForMarketing(true);
      bottomSheetRef.current?.onCloseBottomSheet();
    },
  };

  return (
    <BottomSheet ref={bottomSheetRef}>
      <Text variant={TextVariant.HeadingMD} style={styles.title}>
        {strings('experience_enhancer_modal.title')}
      </Text>
      <View style={styles.content}>
        <Text variant={TextVariant.BodyMD}>
          {strings('experience_enhancer_modal.paragraph1a')}
          <Button
            variant={ButtonVariants.Link}
            label={strings('experience_enhancer_modal.link')}
            // TODO: add link
            onPress={() => {}}
          />
          {strings('experience_enhancer_modal.paragraph1b')}
        </Text>

        <Text variant={TextVariant.BodyMD}>
          {strings('experience_enhancer_modal.paragraph2')}
        </Text>
        <View style={styles.list}>
          <Text style={styles.line}>
            <Text style={styles.dot}>•</Text>{' '}
            {strings('experience_enhancer_modal.bullet1')}
          </Text>
          <Text style={styles.line}>
            <Text style={styles.dot}>•</Text>{' '}
            {strings('experience_enhancer_modal.bullet2')}
          </Text>
          <Text style={styles.line}>
            <Text style={styles.dot}>•</Text>{' '}
            {strings('experience_enhancer_modal.bullet3')}
          </Text>
        </View>
        <Text variant={TextVariant.BodyMD}>
          {strings('experience_enhancer_modal.footer')}
        </Text>
        <BottomSheetFooter
          buttonsAlignment={ButtonsAlignment.Horizontal}
          buttonPropsArray={[cancelButtonProps, acceptButtonProps]}
        />
      </View>
    </BottomSheet>
  );
};

export default ExperienceEnhancerModal;
