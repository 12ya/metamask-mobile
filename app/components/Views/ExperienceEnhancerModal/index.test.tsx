import React from 'react';
import { shallow } from 'enzyme';
import ExperienceEnhancerModal from './';
import { NavigationContainer } from '@react-navigation/native';

describe('ExperienceEnhancerModal', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <NavigationContainer>
        <ExperienceEnhancerModal />
      </NavigationContainer>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
