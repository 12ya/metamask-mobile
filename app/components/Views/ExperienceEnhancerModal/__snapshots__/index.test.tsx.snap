// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExperienceEnhancerModal should render correctly 1`] = `
[
  <Text
    accessibilityRole="text"
    style={
      {
        "color": "#141618",
        "fontFamily": "EuclidCircularB-Bold",
        "fontSize": 18,
        "fontWeight": "700",
        "letterSpacing": 0,
        "lineHeight": 24,
        "marginTop": 15,
        "textAlign": "center",
      }
    }
    testID="experience-enhancer-modal-title"
  >
    Help us enhance your experience
  </Text>,
  <View
    style={
      {
        "display": "flex",
        "flexDirection": "column",
        "gap": 10,
        "margin": 20,
      }
    }
    testID="experience-enhancer-modal-content"
  >
    <Text
      accessibilityRole="text"
      style={
        {
          "color": "#141618",
          "fontFamily": "EuclidCircularB-Regular",
          "fontSize": 14,
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 22,
        }
      }
    >
      In addition to 
      <Text
        accessibilityRole="link"
        accessible={true}
        onPress={[Function]}
        onPressIn={[Function]}
        onPressOut={[Function]}
        style={
          {
            "backgroundColor": "transparent",
            "color": "#141618",
            "fontFamily": "EuclidCircularB-Regular",
            "fontSize": 14,
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 22,
          }
        }
        suppressHighlighting={true}
        testID="experience-enhancer-modal-link-button"
      >
        <Text
          accessibilityRole="text"
          style={
            {
              "color": "#0376c9",
              "fontFamily": "EuclidCircularB-Regular",
              "fontSize": 14,
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 22,
            }
          }
        >
          MetaMetrics
        </Text>
      </Text>
      , we'd like to use data (like information from cookies) to learn how you interact with our marketing communications.
    </Text>
    <Text
      accessibilityRole="text"
      style={
        {
          "color": "#141618",
          "fontFamily": "EuclidCircularB-Regular",
          "fontSize": 14,
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 22,
        }
      }
      testID="experience-enhancer-modal-paragraph2"
    >
      This helps us personalize what we share with you, like:
    </Text>
    <View
      style={
        {
          "display": "flex",
          "gap": 10,
          "marginLeft": 10,
        }
      }
    >
      <Text
        accessibilityRole="text"
        style={
          {
            "color": "#141618",
            "display": "flex",
            "flexDirection": "row",
            "fontFamily": "EuclidCircularB-Regular",
            "fontSize": 14,
            "fontWeight": "400",
            "gap": 10,
            "letterSpacing": 0,
            "lineHeight": 22,
          }
        }
        testID="experience-enhancer-modal-bullet1"
      >
        <Text
          accessibilityRole="text"
          style={
            {
              "color": "#141618",
              "fontFamily": "EuclidCircularB-Regular",
              "fontSize": 14,
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 22,
              "marginRight": 10,
            }
          }
        >
          •
        </Text>
         
        Latest developments
      </Text>
      <Text
        accessibilityRole="text"
        style={
          {
            "color": "#141618",
            "display": "flex",
            "flexDirection": "row",
            "fontFamily": "EuclidCircularB-Regular",
            "fontSize": 14,
            "fontWeight": "400",
            "gap": 10,
            "letterSpacing": 0,
            "lineHeight": 22,
          }
        }
        testID="experience-enhancer-modal-bullet2"
      >
        <Text
          accessibilityRole="text"
          style={
            {
              "color": "#141618",
              "fontFamily": "EuclidCircularB-Regular",
              "fontSize": 14,
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 22,
              "marginRight": 10,
            }
          }
        >
          •
        </Text>
         
        Product features
      </Text>
      <Text
        accessibilityRole="text"
        style={
          {
            "color": "#141618",
            "display": "flex",
            "flexDirection": "row",
            "fontFamily": "EuclidCircularB-Regular",
            "fontSize": 14,
            "fontWeight": "400",
            "gap": 10,
            "letterSpacing": 0,
            "lineHeight": 22,
          }
        }
        testID="experience-enhancer-modal-bullet3"
      >
        <Text
          accessibilityRole="text"
          style={
            {
              "color": "#141618",
              "fontFamily": "EuclidCircularB-Regular",
              "fontSize": 14,
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 22,
              "marginRight": 10,
            }
          }
        >
          •
        </Text>
         
        Other relevant promotional materials
      </Text>
    </View>
    <Text
      accessibilityRole="text"
      style={
        {
          "color": "#141618",
          "fontFamily": "EuclidCircularB-Regular",
          "fontSize": 14,
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 22,
        }
      }
      testID="experience-enhancer-modal-footer"
    >
      Remember, we never sell the data you provide and you can opt out any time.
    </Text>
    <View
      style={
        {
          "backgroundColor": "#ffffff",
          "flexDirection": "row",
          "paddingHorizontal": 8,
          "paddingVertical": 4,
        }
      }
      testID="bottomsheetfooter"
    >
      <TouchableOpacity
        accessibilityRole="button"
        accessible={true}
        activeOpacity={1}
        onPress={[Function]}
        onPressIn={[Function]}
        onPressOut={[Function]}
        style={
          {
            "alignItems": "center",
            "alignSelf": "flex-start",
            "backgroundColor": "transparent",
            "borderColor": "#0376c9",
            "borderRadius": 24,
            "borderWidth": 1,
            "flex": 1,
            "flexDirection": "row",
            "height": 48,
            "justifyContent": "center",
            "paddingHorizontal": 16,
          }
        }
        testID="experience-enhancer-modal-cancel-button"
      >
        <Text
          accessibilityRole="text"
          style={
            {
              "color": "#0376c9",
              "fontFamily": "EuclidCircularB-Regular",
              "fontSize": 14,
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 22,
            }
          }
        >
          No thanks
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        accessibilityRole="button"
        accessible={true}
        activeOpacity={1}
        onPress={[Function]}
        onPressIn={[Function]}
        onPressOut={[Function]}
        style={
          {
            "alignItems": "center",
            "alignSelf": "flex-start",
            "backgroundColor": "#0376c9",
            "borderRadius": 24,
            "flex": 1,
            "flexDirection": "row",
            "height": 48,
            "justifyContent": "center",
            "marginLeft": 16,
            "marginTop": 0,
            "paddingHorizontal": 16,
          }
        }
        testID="experience-enhancer-modal-accept-button"
      >
        <Text
          accessibilityRole="text"
          style={
            {
              "color": "#ffffff",
              "fontFamily": "EuclidCircularB-Regular",
              "fontSize": 14,
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 22,
            }
          }
        >
          I agree
        </Text>
      </TouchableOpacity>
    </View>
  </View>,
]
`;
